servers/Template.Mcp.Server/src/Program.cs

```diff
@@ -38,7 +38,9 @@ private static async Task<int> Main(string[] args)
             var commandFactory = serviceProvider.GetRequiredService<CommandFactory>();
             var rootCommand = commandFactory.RootCommand;
             var parseResult = rootCommand.Parse(args);
-            return await parseResult.InvokeAsync();
+            var status = await parseResult.InvokeAsync();
+
+            return (status >= 200 && status < 300) ? 0 : 1;
         }
```

Comment:
You might be able to put this return logic in a helper and share it amongst the servers.

Reply:
I thought about putting it in a utility class we could share, but did not really find a place I was comforatble with in the current directory structure. Should it live in `core/Microsoft.Mcp.Core`?

/cc @hallipr